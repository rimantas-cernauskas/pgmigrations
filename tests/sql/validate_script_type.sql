-- Test function pgmigrations.validate_script_type()
\i tests/sql/setup/init.sql
BEGIN;
SELECT plan(2);

SET ROLE __pgmigrations_testuser;
SELECT is (
    pgmigrations.validate_script_type( 'up' ),
    TRUE,
    'validate_script_type() should return TRUE when called with existing script type'
);

PREPARE my_query AS
    SELECT pgmigrations.validate_script_type( $1 );

select throws_ok(
    'EXECUTE my_query( ''should not exist'' )',
    'Invalid script type "should not exist"',
    'should throw exception when called with non exitent script type'
);
DEALLOCATE my_query;

SELECT * FROM finish();

ROLLBACK;