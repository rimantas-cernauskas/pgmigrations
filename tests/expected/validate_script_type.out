-- Test function pgmigrations.validate_script_type()
\i tests/sql/setup/init.sql
\set ECHO errors
1..2
ok 1 - validate_script_type() should return TRUE when called with existing script type
ok 2 - should throw exception when called with non exitent script type
